compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -8219,4 +8229,14 @@ private FieldInfo(List<BType> types, boolean required, boolean readonly) {
             this.readonly = readonly;
         }
     }
+
+    private static class TypeSymbolPair {
+        private BVarSymbol fieldSymbol;
+        private BType determinedType;
```

Comment:
Can't we just use `symbol` and `type`?

Reply:
Can. Felt it might be too generic since this was added for a specific purpose.